src/OpenRiaServices.Server/Test/DomainServiceTests.cs

```diff
@@ -582,8 +590,42 @@ public void ObjectContextExtensions_AttachAsModified()
 
             var currentEntry = nw.ObjectContext.ObjectStateManager.GetObjectStateEntry(current);
 
-            string[] changedProperties = currentEntry.GetModifiedProperties().ToArray();
-            Assert.IsTrue(changedProperties.Contains("Description"));
+            string[] actualChangedProperties = currentEntry.GetModifiedProperties().ToArray();
+            
+            CollectionAssert.AreEquivalent(expectedChangedProperties, actualChangedProperties);
+        }
+
+        // Verify that AttachAsModified works correctly when no original values are provided for non-concurrency properties.
+        [TestMethod]
+        public void DbContextEFCoreExtensions_AttachAsModifiedCore()
+        {
+            var expectedChangedProperties = new[]
+            {
+                "CategoryName",
+                "Description",
+                "Picture"
+            };
+            var nw = new TestDomainServices.EFCore.Northwind();
+            DomainServiceContext ctxt = new DomainServiceContext(new MockDataService(), new MockUser("mathew"), DomainOperationType.Submit);
+            nw.Initialize(ctxt);
+
+            var current = new EFCoreModels.Northwind.Category()
+            {
+                CategoryID = 1,
+                CategoryName = "Category",
+                Description = "My category"
+            };
+            var original = new EFCoreModels.Northwind.Category()
+            {
+                CategoryID = 1,
+                CategoryName = "Category"
+            };
+
+            DbContextEFCoreExtensions.AttachAsModified(nw.DbContext.Categories, current, original, nw.DbContext);
+
+            string[] actualChangedProperties = nw.DbContext.Entry(current).Properties.Where(p => p.IsModified).Select(p => p.Metadata.Name).ToArray();
+
+            CollectionAssert.AreEquivalent(expectedChangedProperties, actualChangedProperties);
```

Comment:
To be sure that the code works we should make sure that we add a separate test with a type having the RoundtripOriginalAttribute applied (in which case CategoryName should not be modified) and maybe some default value such as int?

Does it make sens to also include variation with Exclude attribute ?

Reply:
we could use "Product" which has roundtrip attribute on "Weight" and exclude on SafetyStockLevel

- so having equal `Weight` that property should "not" be modified
- it should not mark SafetyStockLevel as modified (if equal)

Reply:
Looked into it and added tests